### PR TITLE
add bebop trades

### PR DESCRIPTION
**Pull request by B1boid**
Bebop - https://bebop.xyz

Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by Hosuke** [comment]
Hi @B1boid, you can add some seed records for bebop trades in this file:
https://github.com/duneanalytics/spellbook/blob/main/seeds/dex/trades/dex_trades_seed.csv

**Comment by B1boid** [comment]
Hey @Hosuke 
I see dbt-test failed because of the error:
[UNRESOLVED_ROUTINE] Cannot resolve function `json_extract` on search path [`system`.`builtin`, `system`.`session`, `hive_metastore`.`default`].; line 29 pos 6

But this request with `json_extract` works fine on Dune: https://dune.com/queries/2342200

if it's a blocker, how can i fix this problem?

**Comment by Hosuke** [comment]
> Hey @Hosuke I see dbt-test failed because of the error: [UNRESOLVED_ROUTINE] Cannot resolve function `json_extract` on search path [`system`.`builtin`, `system`.`session`, `hive_metastore`.`default`].; line 29 pos 6
> 
> But this request with `json_extract` works fine on Dune: https://dune.com/queries/2342200
> 
> if it's a blocker, how can i fix this problem?

Currently the spellbook is using spark sql engine for compilation, we may need to find an equivalent function for replacement. (Maybe in future we can directly merge dune sql syntax into spellbook)

Maybe this one:
https://docs.databricks.com/sql/language-manual/functions/get_json_object.html

**Comment by B1boid** [comment]
Hey @Hosuke 
Now tests failed because of duplication, I think this may be caused by the issue:
https://github.com/duneanalytics/spellbook/issues/3107

That is, we will need to wait for this bug to be fixed, or can we merge this PR now, if everything is ok?

**Comment by Hosuke** [comment]
Seems so:
https://dune.com/queries/2351933?d=1

> Hey @Hosuke Now tests failed because of duplication, I think this may be caused by the issue: #3107
> 
> That is, we will need to wait for this bug to be fixed, or can we merge this PR now, if everything is ok?

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @B1boid for the great work.
Sorry we took some time to resolve the dupes issues.

- Metadata & naming conventions ✅
- Config blocks match uniqueness tests in schema files ✅
- Tests and seeds added for the new models ✅
- Incremental filters added when joining large tables ✅
- Contract addresses are lowercase ✅
- Avoid joining to large tables(*.transactions) as possible ✅
- `CAST()` used in the base models ✅

[Sample bebop ethereum trades](https://dune.com/queries/2358398)
[Sample bebop arbitrum trades](https://dune.com/queries/2352349)
[Sample bebop polygon trades](https://dune.com/queries/2358406)

Also we may recycle the DuneSQL syntax spells in historical commits from PR in future upgrade.

**Comment by jeff-dude** [comment]
@B1boid thank you for the contribution. one question: when i see the usage of call function decoded tables & applying values to `trace_address`, it makes me think this may belong in `dex_aggregator.trades`, rather than `dex.trades`? can you help confirm if bebop is an aggregator or not?

**Comment by B1boid** [comment]
> @B1boid thank you for the contribution. one question: when i see the usage of call function decoded tables & applying values to `trace_address`, it makes me think this may belong in `dex_aggregator.trades`, rather than `dex.trades`? can you help confirm if bebop is an aggregator or not?

Yeah, Bebop is an aggregator, so I pushed a fix, thanks!

File: seeds/dex/aggregator/dex_aggregator_seed.csv
Comment (Hosuke) [inline seeds/dex/aggregator/dex_aggregator_seed.csv:59]: Hi @B1boid, you can fill in the `trace_address` field in this seed file as well.

Comment (B1boid) [reply seeds/dex/aggregator/dex_aggregator_seed.csv]: Thanks, done ✅

**Review by jeff-dude — APPROVED** [review]
amazing -- great work @B1boid and @Hosuke 🙌

**Pull request merged**
